packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -287,13 +294,449 @@ class HeaderMatchImpl extends HttpHeaderMatch {
+
+  /**
+   * The value of the metadata with the given name in the request must not end
+   * with the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param suffix The suffix to test against
+   */
+  static valueDoesNotEndWith(metadataName: string, suffix: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, { suffix });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must include
+   * the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param regex The regex to test against
+   */
+  static valueMatchesRegex(metadataName: string, regex: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, false, { regex });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must not
+   * include the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param regex The regex to test against
+   */
+  static valueDoesNotMatchRegex(metadataName: string, regex: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, { regex });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must be in a
+   * range of values.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsInRange(metadataName: string, start: number, end: number): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, false, {
+      range: {
+        start,
+        end,
+      },
+    });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must not be in
+   * a range of values.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsNotInRange(metadataName: string, start: number, end: number): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, {
+      range: {
+        start,
+        end,
+      },
+    });
+  }
+
+  /**
+   * Returns the metadata match configuration.
+   */
+  abstract bind(scope: Construct): GrpcRouteMetadataMatchConfig;
+}
+
+class MetadataMatchImpl extends GrpcMetadataMatch {
+  constructor(
+    private readonly metadataName: string,
+    private readonly invert: boolean,
+    private readonly matchProperty: CfnRoute.GrpcRouteMetadataMatchMethodProperty,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): GrpcRouteMetadataMatchConfig {
+    return {
+      metadataMatch: {
+        name: this.metadataName,
+        invert: this.invert,
+        match: this.matchProperty,
+      },
+    };
+  }
```

Comment:
Let's move `GrpcMetadataMatch`, `HttpHeaderMatch`,  `HttpRouteMatchMethod`, `HttpRoutePathOrPrefixMatch`, and `QueryParameterMatch` to their own file (since they are used in multiple places).

Reply:
Well noted.